git/github/github.go

```diff
@@ -302,6 +317,83 @@ func (c *Client) GetCheckRunID(ctx context.Context, owner string, repo string, S
 	return nil, nil
 }
 
+// GetAllCheckRunsForRef returns all existing GitHub CheckRuns if a match for the Owner, Repo, SHA, and appID.
+func (c *Client) GetAllCheckRunsForRef(ctx context.Context, owner string, repo string, SHA string, appID int64) ([]*ghapi.CheckRun, error) {
+	filter := "all"
+
+	res, _, err := c.GetChecksService().ListCheckRunsForRef(
+		ctx,
+		owner,
+		repo,
+		SHA,
+		&ghapi.ListCheckRunsOptions{
+			AppID:  &appID,
+			Filter: &filter,
+		},
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if *res.Total == 0 {
+		c.logger.Info("Found no CheckRuns for the ref", "SHA", SHA)
+		return nil, nil
+	}
+
+	return res.CheckRuns, nil
+}
+
+// GetCheckRunIDAndIfUpdateNeeded returns an existing GitHub CheckRun ID if a match is found and if updated/create is needed for the SHA, externalID and appID.
+func (c *Client) GetCheckRunIDAndIfUpdateNeeded(checkRuns []*ghapi.CheckRun, checkRun *CheckRunAdapter) (*int64, bool, error) {
```

Comment:
This should be 2 functions to comply with single responsibility principle:
* GetCheckRun
* IsupdateNeeded

Reply:
fixed.